transactions/demo-typescript/demo-rts.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+
+# DESCRIPTION
+#
+# This shell script does the following.
+#   1. Starts `local-cluster` (plutip) where logs are dumped to to stderr and
+#      `./local-cluster.log`.
+#      Note that we keep plutip's `./local-cluster-info.json`
+#   2. Starts `ogmios` on host 127.0.0.1 with port 1337 where logs are dumped
+#      to stderr and `ogmios.log`
+#   3. Dumps a config file (compatible with the the demo Typescript
+#      application) to `./demo-rtsconfig.json` and stdout.
+#
+#  Note that the config file is dumped to stdout, and is the only thing that is
+#  dumped to stdout, so blocking reads on stdout can be used to detect when
+#  everything is ready)
+#
+# ENVIRONMENT
+#   - `ADA`: the total number of ADA per UTxO to start the cluter with
+#   (default: 69420). See [1].
+#   - `UTXOS`: the total number of UTxOs with amount `$ADA`. (default: 1). See
+#     [1]
+#
+# REFERENCES
+# [1]: Plutip's `local-cluster` CLI reference:
+#      https://github.com/mlabs-haskell/plutip/blob/master/local-cluster/README.md
+
+export ADA="${ADA:-69420}"
+export UTXOS="${UTXOS:-1}"
+
+# Ensure we kill all the background processes (plutip + ogmios) when we exit
+trap 'trap - EXIT && kill -- -$$ && exit 0' EXIT
+
+###########################
+# Setting up the plutip cluster
+###########################
+export WALLET_DIR
+WALLET_DIR="$(mktemp -d)"
+export NODE_WORKING_DIR
+NODE_WORKING_DIR="$(mktemp -d)"
+
+# The directory for plutip's local cluster information see [1]
+export LOCAL_CLUSTER_INFO_JSON="./local-cluster-info.json"
+
+# Ensure $LOCAL_CLUSTER_INFO_JSON exists because plutip doesn't like it if this
+# file doesn't already exist.
+touch "$LOCAL_CLUSTER_INFO_JSON"
```

Comment:
What? That doesn't sound right and I implemented the file output :D

Reply:
Yeah that doesn't seem right. It works even if the file doesn't already exist!